src/app/app.component.ts

```diff
@@ -1,276 +1,338 @@
 import { Component, OnInit } from '@angular/core';
-import { HttpClient } from '@angular/common/http';
-import { NgForm } from '@angular/forms';
+import { Mailbox, NaCl, ZaxParsedMessage, ZaxMessageKind, ZaxFileMessage } from '@vault12/glow.ts';
+
+// Glow type extensions to represent data downloaded from the relay conveniently
+type MessageView = ZaxParsedMessage & { isSelected?: boolean };
+interface MailboxView extends Mailbox {
+  counter?: number;
+  messages?: MessageView[];
+  recipients?: string[];
+}
+
+enum UIAction {
+  keyAdded,
+  mailboxCreated,
+  messageSent,
+  fileSent,
+  messagesLoading,
+  counterLoading
+}
 
 @Component({
   selector: 'app-root',
   templateUrl: './app.component.html',
   styleUrls: ['./app.component.css']
 })
 export class AppComponent implements OnInit {
-  private mailboxPrefix = '_mailbox';
-
-  // Glow instances and mailboxes
-  private glow;
-  private relay;
-  private mailbox;
-  activeMailbox = null;
-  mailboxes = [];
-
-  // UI flags
-  showRefreshLoader = false;
-  showMessagesLoader = false;
-  messageSent = false;
-  keyAdded = false;
-  isEditing = false;
+  // Needed to find in local storage and load all previously generated mailboxes
+  private readonly mailboxPrefix = '_mailbox';
+
+  // Cached mailbox and file data
+  mailboxes: MailboxView[] = [];
+  activeMailbox: MailboxView = null;
+  selectedFile: File;
+
+  ZaxMessageKind = ZaxMessageKind;
 
   // UI defaults
+  UIAction = UIAction;
+  UIFlags = {
+    [UIAction.keyAdded]: false,
+    [UIAction.mailboxCreated]: false,
+    [UIAction.messageSent]: false,
+    [UIAction.fileSent]: false,
+    [UIAction.messagesLoading]: false,
+    [UIAction.counterLoading]: false
+  };
```

Comment:
why do we need this flags why not using simple properties?

Reply:
`showBadge(action)` method is used to display a "success toast" after actions like "send message" or "add key", so I wanted to handle this UI logic in a single method. Otherwise, I'd have to put the same `setTimeout` in every method. Do you have better ideas?

Reply:
it's ok for badges but there are also messagesLoading, counterLoading which are not used in showBadge

Reply:
Ok, I'll change it in the next PR